File: cf940b.go
package main

import (
	"fmt"
	"io"
	"os"
)

func cf940b(in io.Reader, out io.Writer) {
	var n, k, a, b, res int
	fmt.Fscan(in, &n, &k, &a, &b)
	for n > 1 {
		res += n % k * a
		n -= n % k
		if (n-n/k)*a < b {
			res += (n - 1) * a
			break
		}
		res += b
		n /= k
	}
	fmt.Fprint(out, res)
}

func main() {
	cf940b(os.Stdin, os.Stdout)
}

File: cf940b.py
n = int(input())
k = int(input())
A = int(input())
B = int(input())

res = 0
while n > 1:
    res += (n % k) * A
    n -= n % k
    if (n - n // k) * A <= B:
        res += (n - 1) * A
        break
    res += B
    n //= k
print(res)